app/src/main/java/de/danoeh/antennapod/ui/episodeslist/FeedItemMenuHandler.java

```diff
@@ -169,6 +181,66 @@ public static void setItemTitle(Menu menu, int id, int noMedia) {
         }
     }
 
+    /**
+     * This method checks if the selected items are allowed to be moved to the top, bottom or both of the Queue.
+     * @param queue The FeedItems currently in the Queue.
+     * @param selectedItems The FeedItems for which the check is performed.
+     * @return A boolean array where:
+     *           [0] is true if moving to the top is allowed, false otherwise.
+     *           [1] is true if moving to the bottom is allowed, false otherwise.
+     * */
+    public static boolean[] canMove(List<FeedItem> queue, List<FeedItem> selectedItems) {
```

Comment:
This is super specific to the queue. Maybe this could be done from the outside just for the queue, calling `menu.findItem(...).setVisible(...)` directly? This would also solve passing the queue (which is null in almost all cases) to the utility method